kedro-datasets/tests/pandas/test_csv_dataset.py

```diff
@@ -92,6 +93,16 @@ def test_save_and_load(self, csv_data_set, dummy_dataframe):
         reloaded = csv_data_set.load()
         assert_frame_equal(dummy_dataframe, reloaded)
 
+    def test_preview(self, csv_data_set, dummy_dataframe):
+        """Test _preview returns the correct nrows amount."""
+        nrows = 2
+
+        csv_data_set.save(dummy_dataframe)
+        response = csv_data_set._preview(nrows=nrows)
+
+        for rows in response.values():
```

Comment:
hey @AntonyMilneQB so response returns a dict which for example, looks like this 

```
{
            "id": {0: "id1", 1: "id2", 2: "id3"},
            "companyRating": {0: "100%", 1: "67%", 2: "67%"},
            "companyLocation": {0: "London", 1: "Paris", 2: "New York"},
     }
```

So my test is basically saying if i define nrows = 2, i expect it should only return to 2 values per key, eg

```
{
            "id": {0: "id1", 1: "id2","},
            "companyRating": {0: "100%", 1: "67%"},
            "companyLocation": {0: "London", 1: "Paris"},
    }
```

Reply:
ok, got it, thanks. I think there's two improvements we need to make here. Happy to show you more tomorrow but let me just point you in the right direction for now...

1. we should have a stricter test since just checking that `len` is correct doesn't really test much here. e.g. I could have generated `{"rubbish": "ab"}` and it would still pass the test, even though it's completely the wrong result for `_preview`. We should explicitly test that the returned dictionary itself actually matches what we expect (like `assert response =  ...`)
2. we should test more than one case for `nrows` and also what happens when it's not supplied as an argument (so `nrows=None`): check `@pytest.mark.parametrize` for how you can do this